### PR TITLE
js: Allow Acking of JS messages without JS context

### PR DESCRIPTION
**Pull request by wallyqs**
This would allow regular subscribers to `Ack()` messages with a reply from JetStream.

Signed-off-by: Waldemar Quevedo <wally@synadia.com>

### CODE REVIEW

File: test/js_test.go [hunk 1]
```diff
@@ -970,6 +986,27 @@ func TestJetStreamImportDirectOnly(t *testing.T) {
 		}
 	}
 
+	// Do push based consumer using a regular NATS subscription on the import subject.
+	sub, err = nc.SubscribeSync("p.d3")
```
Comment (wallyqs) [inline test/js_test.go:990]: Regular subscribe here is the same as this:
```
sub, err = js.SubscribeSync("ORDERS", nats.PushDirect("p.d3"))
```
so with this change we could remove `PushDirect` as well potentially.

Comment (derekcollison) [reply test/js_test.go:990]: I think we should.

**Review by derekcollison — APPROVED** [review]
LGTM

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.

**Pull request merged**